src/main/java/by/exadel/internship/controller/FormController.java

```diff
@@ -43,9 +42,14 @@ public FormFullDTO addNewForm(@RequestPart(name = "form") FormRegisterDTO form,
 
     @AdminAccessControl
     @GetMapping
-    @ApiOperation("Get all forms by internship id")
-    public List<FormFullDTO> getAllFormsByInternshipId(@RequestParam("internshipId") UUID internshipId) {
-        return formService.getAllByInternshipId(internshipId);
+    @ApiOperation("Get all forms by internship id or user id")
+    public List<FormFullDTO> getAllFormsById(@RequestParam(value = "internshipId", required = false) UUID internshipId
+            , @RequestParam(value = "userId", required = false) UUID userId) {
+
+        return internshipId != null
```

Comment:
Trusty speaking I don't like this approach. On the other hand my suggested optimization can overcomplicate code.
Today I can suggest if endpoint works  only one from 2 options (search by userId or intershipId) in case both variable are null or not empty you throw exception.
@SergeyKoval @viktarkolbik your thoughts?

Reply:
in case both variable we get form by internship id

Reply:
> in case both variable we get form by internship id

Ignoring one variable is strange and unexpected behavior.

Reply:
I believe it is good candidate for the dynamic query. Take a look at this approach